osmdroid-shape/src/main/java/org/osmdroid/shape/ShapeMetadataReader.java

```diff
@@ -0,0 +1,26 @@
+package org.osmdroid.shape;
+
+import net.iryndin.jdbf.core.DbfRecord;
+
+import org.osmdroid.util.BoundingBox;
+
+import java.nio.charset.Charset;
+import java.text.ParseException;
+
+public abstract class ShapeMetadataReader {
```

Comment:
I think methods `setBoundingBox` and `getSubDescription` are useful.
Developers might wanna change sub-descriptions if the auto generated ones have wrong values due to the double precision rounding issue. They may also want to change the decimal places.

For example, our project:

![20200418_160752-COLLAGE](https://user-images.githubusercontent.com/14086980/79632042-626f7700-818f-11ea-883e-0e312bcf5ea6.jpg)

Reply:
OK, I see the use case.
But then, it's like the `int fillColor` vs. `Paint fillPaint` competition: why do we use `BoundingBox` but not the number of points? Or other polygon details? Or why is do we use subDescription/BoundingBox for Polygons, but not for Polylines?

I think I've got the solution: instead of a `ShapeMetadataReader` class/interface, we should use something like that:
````java
public interface ShapeMetaSetter{
    void set(DbfRecord metadata, Polygon polygon);
    void set(DbfRecord metadata, Polyline polyline);
    void set(DbfRecord metadata, Marker marker);
}
````
and implement it with a `DefaultShapeMetaSetter` class, where you can do what you want with the metadata AND the bounding box or the number of points, whatever, for the title, the snippet and the sub description.

Do you get it so far?
Because I have an even better idea: why couldn't we also set the paint here?
What's the point? Well, maybe you are able to see in the metadata that this polygon is important (and paint it in red instead of black), or that this polygon's bounding box is tiny, therefore you need to give it a slightly bigger stroke width. Or you want the polygons and the polylines to have a different color, which is not possible for the moment.

That would be brillant, wouldn't it?

Reply:
> OK, I see the use case.
> But then, it's like the `int fillColor` vs. `Paint fillPaint` competition: why do we use `BoundingBox` but not the number of points? Or other polygon details? Or why is do we use subDescription/BoundingBox for Polygons, but not for Polylines?
> 
> I think I've got the solution: instead of a `ShapeMetadataReader` class/interface, we should use something like that:
> 
> ```java
> public interface ShapeMetaSetter{
>     void set(DbfRecord metadata, Polygon polygon);
>     void set(DbfRecord metadata, Polyline polyline);
>     void set(DbfRecord metadata, Marker marker);
> }
> ```
> 
> and implement it with a `DefaultShapeMetaSetter` class, where you can do what you want with the metadata AND the bounding box or the number of points, whatever, for the title, the snippet and the sub description.
> 
> Do you get it so far?
> Because I have an even better idea: why couldn't we also set the paint here?
> What's the point? Well, maybe you are able to see in the metadata that this polygon is important (and paint it in red instead of black), or that this polygon's bounding box is tiny, therefore you need to give it a slightly bigger stroke width. Or you want the polygons and the polylines to have a different color, which is not possible for the moment.
> 
> That would be brillant, wouldn't it?

Good idea. That's right.